src/datasets-context-json.c

```diff
@@ -117,6 +117,7 @@ static json_t *GetSubObjectByKey(json_t *json, const char *key)
         const char *dot = strchr(current_key, '.');
 
         size_t key_len = dot ? (size_t)(dot - current_key) : strlen(current_key);
+        DEBUG_VALIDATE_BUG_ON(key_len > SIG_JSON_CONTENT_KEY_LEN);
```

Comment:
I think I would feel a lot better if at the start of the function we had a generic `if (strlen(key) >= SIG_JSON_CONTENT_KEY_LEN) return NULL;`

Reply:
makes sense